specs/protocol/derivation.md

```diff
@@ -363,29 +363,47 @@ where:
 - `is_last` is a single byte with a value of 1 if the frame is the last in the channel, 0 if there are frames in the
   channel. Any other value makes the frame invalid (it must be ignored by the rollup node).
 
-[batcher-spec]: batching.md
-
 ### Channel Format
 
-A channel is encoded as `channel_encoding`, defined as:
+A channel is encoded by applying a streaming compression algorithm to a list of batches:
 
 ```text
 rlp_batches = []
 for batch in batches:
     rlp_batches.append(batch)
-channel_encoding = compress(rlp_batches)
 ```
 
 where:
 
 - `batches` is the input, a sequence of batches byte-encoded as per the next section ("Batch Encoding")
 - `rlp_batches` is the concatenation of the RLP-encoded batches
-- `compress` is a function performing compression, using the ZLIB algorithm (as specified in [RFC-1950][rfc1950]) with
-  no dictionary
-- `channel_encoding` is the compressed version of `rlp_batches`
+
+Prior to the Fjord upgrade,
+
+```text
+channel_encoding = zlib_compress(rlp_batches)
+```
+
+where zlib_compress is the ZLIB algorithm (as specified in [RFC-1950][rfc1950]) with no dictionary.
 
 [rfc1950]: https://www.rfc-editor.org/rfc/rfc1950.html
 
+After the Fjord upgrade, there are two possible channel encoding formats. The legacy format,
+which is exactly as above, and a versioned format:
+
+```text
+channel_encoding = `version_byte ++ compress(rlp_batches)`
+```
+
+For the versioned format, the `version_byte` must never have its 4 lower order bits set to 0x1000,
+as those bits indicate the channel encoding is of legacy type by definition of the zlib encoding
+header.
+
+As of the Fjord upgrade, the only valid `version_byte` is `1`, which indicates `compress()` is the
+Brotli compression algorithm (as specified in [RFC-7932][rfc7932]) with no custom dictionary.
```

Comment:
Wondering if we should also update the `Channel Reader (Batch Decoding)` section to mention how we check for zlib vs brotli?

Reply:
Hmm, that section only seemed to reference the batch-format section which does seem incomplete.  I added a reference to the channel-encoding format as well. Should be clear from the way we note the 0x0001 low order bits can be used for disambiguation how one would determine zlib vs brotli?

Reply:
Yeah sounds good